code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +122,296 @@
+				/													\
+			Putrid													Crimson
+	Eaten too much, died too little								Eaten too little, died too much
+	*/
+	if(istype(src, /mob/living/simple_animal/hostile/necro/zombie/turned))
+		if(times_revived > 0 || times_eaten > 0)
+			evolve(/mob/living/simple_animal/hostile/necro/zombie/rotting)
+
+	else if (istype(src, /mob/living/simple_animal/hostile/necro/zombie/rotting))
+		if(times_eaten > (1+times_revived)*2) //Have to have eaten at least twice and more than double that of times died
+			evolve(/mob/living/simple_animal/hostile/necro/zombie/putrid)
+		else if(times_revived > times_eaten+1) //Died at least twice
+			evolve(/mob/living/simple_animal/hostile/necro/zombie/crimson)
+
+	busy = 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/evolve(var/mob/living/simple_animal/evolve_to)
+	if(ispath(evolve_to, /mob/living/simple_animal/hostile/necro))
+		var/mob/living/evolution = new evolve_to(src.loc,,)
+		evolution.name = name //We want to keep the name
+		if(mind)
+			mind.transfer_to(evolution) //Just in the offchance we have a player in control
+		qdel(src)
+	else
+		//Now, how did you get here when this is supposed to be the zombie evolution tree?
+		new evolve_to(src.loc)
+		qdel(src)
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/drop_host()
+
+/mob/living/simple_animal/hostile/necro/zombie/delayedRegen()
+	..()
+	times_revived += 1
+
+/mob/living/simple_animal/hostile/necro/zombie/turned //Not very useful
+	icon_state = "zombie_turned" //Looks almost not unlike just a naked guy to potentially catch others off guard
+	icon_living = "zombie_turned"
+	icon_dead = "zombie_turned_dead"
+	desc = "A reanimated corpse that looks like it has seen better days. This one still appears quite human."
+	maxHealth = 50
+	health = 50
+	can_evolve = 1
+	var/mob/living/carbon/human/host //Whoever the zombie was previously, kept in a reference to potentially bring back
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/drop_host()
+	qdel(host) //Bye bye
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/gib()
+	..()
+	drop_host()
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/dust()
+	..()
+	drop_host()
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/attackby(var/obj/item/weapon/W, var/mob/user)
+	..()
+	if(stat == DEAD) //Can only attempt to unzombify if they're dead
+		if(istype (W, /obj/item/weapon/storage/bible)) //This calls for divine intervention
+			var/obj/item/weapon/storage/bible/bible = W
+			visible_message("\The [user] begins whacking at [src] repeatedly with a bible for some reason.")
+			to_chat (user, "<span class='notice'>You attempt to invoke the power of [bible.deity_name] to bring this poor soul back from the brink.</span>")
+
+			var/chaplain = 0 //Are we the Chaplain ? Used for simplification
+			if(user.mind && (user.mind.assigned_role == "Chaplain"))
+				chaplain = 1 //Indeed we are
+
+			spawn(25) //So there's a nice delay
```

Comment:
Do_after's good here though, because it's a player doing it not a mob, so they won't wander